extensions/micrometer/deployment/src/test/java/io/quarkus/micrometer/deployment/binder/NettyMetricsTest.java

```diff
@@ -0,0 +1,150 @@
+            .overrideConfigKey("quarkus.redis.devservices.enabled", "false");
+
+    @Inject
+    @Any
+    Instance<MeterBinder> binders;
+
+    @Inject
+    MeterRegistry registry;
+
+    @Inject
+    Vertx vertx;
+
+    private static final Set<Tag> NAM_PBBA_TAGS = Tags.of(
+            "id", String.valueOf(PooledByteBufAllocator.DEFAULT.hashCode()),
+            "allocator.type", "PooledByteBufAllocator")
+            .stream()
+            .collect(Collectors.toSet());
+
+    private static final Set<Tag> NAM_UNPBBA_TAGS = Tags.of(
+            "id", String.valueOf(UnpooledByteBufAllocator.DEFAULT.hashCode()),
+            "allocator.type", "UnpooledByteBufAllocator")
+            .stream()
+            .collect(Collectors.toSet());
+
+    private static final Set<Tag> VX_NAM_PBBA_TAGS = Tags.of(
+            "id", String.valueOf(VertxByteBufAllocator.POOLED_ALLOCATOR.hashCode()),
+            "allocator.type", "PooledByteBufAllocator")
+            .stream()
+            .collect(Collectors.toSet());
+
+    private static final Set<Tag> VX_NAM_UNPBBA_TAGS = Tags.of(
+            "id", String.valueOf(VertxByteBufAllocator.UNPOOLED_ALLOCATOR.hashCode()),
+            "allocator.type", "UnpooledByteBufAllocator")
+            .stream()
+            .collect(Collectors.toSet());
+
+    private void testNettyMetrics(long expected, Class<? extends MeterBinder> mbClass) {
+        Assertions.assertFalse(binders.isUnsatisfied());
+        long count = binders.stream()
+                .filter(mbClass::isInstance)
+                .count();
+        Assertions.assertEquals(expected, count);
+    }
+
+    @Test
+    public void testNettyAllocatorMetrics() {
+        testNettyMetrics(5L, NettyAllocatorMetrics.class);
+    }
+
+    @Test
+    public void testNettyEventExecutorMetrics() {
+        testNettyMetrics(2L, NettyEventExecutorMetrics.class);
+    }
+
+    @Test
+    public void testAllocatorMetricsValues() {
+        RestAssured.get("/hello/Netty").then().body(Matchers.equalTo("hello Netty"));
+
+        boolean pbba_found = false;
+        boolean unpbba_found = false;
+        boolean vx_pbba_found = false;
+        boolean vx_unpbba_found = false;
+        List<Meter> meters = registry.getMeters();
+        for (Meter meter : meters) {
+            List<Tag> tags = meter.getId().getTags();
+            if (tags.containsAll(NAM_PBBA_TAGS)) {
+                pbba_found = true;
+            }
+            if (tags.containsAll(NAM_UNPBBA_TAGS)) {
+                unpbba_found = true;
+            }
+            if (tags.containsAll(VX_NAM_PBBA_TAGS)) {
+                vx_pbba_found = true;
+            }
+            if (tags.containsAll(VX_NAM_UNPBBA_TAGS)) {
+                vx_unpbba_found = true;
+            }
+        }
+        Assertions.assertTrue(pbba_found);
+        Assertions.assertTrue(unpbba_found);
+        Assertions.assertTrue(vx_pbba_found);
+        Assertions.assertTrue(vx_unpbba_found);
```

Comment:
If the client request is the very first one happening from where the server is started you could assert that the used memory values have changed before/after too